ipld/bitfield/src/lib.rs

```diff
@@ -367,7 +406,20 @@ impl BitAnd<&BitField> for &BitField {
 
     #[inline]
     fn bitand(self, rhs: &BitField) -> Self::Output {
-        BitField::from_ranges(self.ranges().intersection(rhs.ranges()))
+        if self.is_trivially_empty() || rhs.is_trivially_empty() {
+            BitField::new()
+        } else {
+            BitField::from_ranges(self.ranges().intersection(rhs.ranges()))
+        }
+    }
+}
+
+impl BitAnd<BitField> for BitField {
```

Comment:
Implementing by-value ops encourages users to perform by-value ops where possible (possibly leading to optimizations).

Reply:
This uses `impl BitAnd<&BitField> for &BitField {` under the hood so there's no speedup to using this instead correct?  But you're saying just letting them work with values directly can lead to user side optimizations?

Reply:
Which is what I’ve been doing recently. 😂